entity.go

```diff
@@ -85,12 +109,16 @@ func NewMultipart(header Header, parts []*Entity) (*Entity, error) {
 // error that verifies IsUnknownCharset or IsUnknownEncoding, but also returns
 // an Entity that can be read.
 func Read(r io.Reader) (*Entity, error) {
-	br := bufio.NewReader(r)
+	lr := &limitedReader{R: r, N: maxHeaderBytes}
+	br := bufio.NewReader(lr)
+
```

Comment:
maybe a naive question, but why limit the size at all?

We could simply defer this to the caller no?  
I get that if you are implementing a server that you definitely care about allocations and DOS attacks.

However in the case of a client sanity checks are performed by the server and people expects that a client just reads a mail if it exists on the server / filesystem no matter how large it is. Obviously theit policy allowed for it else they wouldn't have received it

Reply:
The caller can't easily limit the size of the header and not the size of the body. The size of the header matters because the whole header is kept in memory (that's not the case for the body).

Reply:
I see, makes sense